src/python/pants/init/target_roots_calculator.py

```diff
@@ -127,11 +127,17 @@ def create(cls, options, session, symbol_table, build_root=None):
     # initialization paths.
     changed_options = options.for_scope('changed')
     changed_request = ChangedRequest.from_options(changed_options)
-
+    owned_files = options.for_global_scope().owner_of
     logger.debug('spec_roots are: %s', spec_roots)
     logger.debug('changed_request is: %s', changed_request)
     scm = get_scm()
     change_calculator = ChangeCalculator(session, symbol_table, scm) if scm else None
+    owner_calculator = OwnerCalculator(session, symbol_table) if owned_files else None
+    logger.debug('owner_files are: %s', owned_files)
+
+    if changed_request.is_actionable() and owned_files:
```

Comment:
It would be good to unify checking of "exactly one of these three types of options" into one check here, with one large and helpful error message (not a paragraph probably, but a few sentences certainly explaining the three choices that exist). You could then remove the `if spec_roots` checks from the codepaths below.